src/ecc_key.c

```diff
@@ -1276,7 +1277,13 @@ int ec_key_sec2prot(struct zpc_ec_key *ec_key, enum ec_key_sec sec)
 	io.pkeylen = sizeof(ec_key->prot.protkey);
 	io.pkey = (unsigned char *)&ec_key->prot.protkey;
 
-	rc = ioctl(pkeyfd, PKEY_KBLOB2PROTK3, &io);
+	for (i = 0; i < 10; i++) {
+		rc = ioctl(pkeyfd, PKEY_KBLOB2PROTK3, &io);
```

Comment:
Yes this one is a valid case where the retry is needed.

There are additional places in the code where a secure key is transformed into a protected key. These other places uses PKEY_KBLOB2PROTK2. You will need the retry for this IOCTL as well.

Reply:
Yep, added more loops for sec2prot.